background_scripts/idem_bg.js

```diff
@@ -1,266 +1,353 @@
-
-
 let _ws, _twitch, _tabId, _ide, _log
+let _cg_id
+const SKIP_IDE_DETECTION = false
 const _buffer = []
+const TOC_URL = 'https://tournament-of-code.osc-fr1.scalingo.io/'
+let _current_tournament
+let _joined_tournament
+
+function fetch_toc_api (path, method = 'GET', payload) {
+  return fetch(`${TOC_URL}/api${path}`, {
+    method,
+    headers: { 'Content-Type': 'application/json' },
+    body: !!payload ? JSON.stringify(payload) : undefined
+  }).then(response => response.json())
+}
 
 const actions = {
+  log_activate: function () {
+    _log = true
+    if (!_tabId) return
+    browser.tabs.sendMessage(_tabId, {
+      command: 'log_activate'
+    })
+  },
 
-	log_activate: function() {
-		_log = true
-		if (!_tabId) return
-		browser.tabs.sendMessage(_tabId, {
-			command: "log_activate",
-		})
-	},
+  log_deactivate: function () {
+    _log = false
+    if (!_tabId) return
+    browser.tabs.sendMessage(_tabId, {
+      command: 'log_deactivate'
+    })
+  },
 
-	log_deactivate: function() {
-		_log = false
-		if (!_tabId) return
-		browser.tabs.sendMessage(_tabId, {
-			command: "log_deactivate",
-		})
-	},
+  log_get_status: function () {
+    return _log
+  },
 
-	log_get_status: function() {
-		return _log
-	},
+  tab_get_id: function () {
+    return _tabId
+  },
 
-	tab_get_id: function() {
-		return _tabId
-	},
+  tab_initialize: function () {
+    // init for this tab
+    browser.tabs.query({ currentWindow: true, active: true }).then(tabs => {
+      _tabId = tabs[0].id
+      _ide = SKIP_IDE_DETECTION || tabs[0].url.includes('codingame.com/ide/')
+      set_idem_logo({ tabId: _tabId })
+    });
+    // get and save CG id
+    ensureScriptLoaded().then(() => {
```

Comment:
Le problème du 'marche pas le 1e coup' est ici au milieu de toutes les promises => asynchrone.
Pas sûr que ça soit le meilleur endroit pour faire la récupération des ids mais je sais pas qu'est-ce qui serait le mieux...

```suggestion
    browser.tabs.query({ currentWindow: true, active: true }).then(tabs => {
      _tabId = tabs[0].id
      _ide = SKIP_IDE_DETECTION || tabs[0].url.includes('codingame.com/ide/')
      set_idem_logo({ tabId: _tabId })
      // get and save CG id
      ensureScriptLoaded().then(() => { ...
    });
```